### DISCUSSION TITLE
Discrepancy in TSS enrichment score between cellranger and Signac: am I using the wrong annotation?

### DISCUSSION THREAD

**Discussion by MarkoBarovic** [stuart-lab/signac#1682] [Q&A]
Hi!
First of all thank you for the great documentation and vignettes!

I have a scATACseq project for which I got the cellranger output including the HTML summary. In that summary, GRCm39.e104 is stated as reference. The cellranger output of the TSS enrichment score reports it as 10.4, while the mean of the TSS enrichment score as calculated by `TSSEnrichment()` is at 1.4 (median is also 1.4). I am using the workflow as in the vignette for human PBMCs, adapted to mouse (EnsDb.Mmusculus.v79 as reference annotation). Am I missing something? Should I be using a different reference annotation?

Thanks a lot in advance!

**Comment by MarkoBarovic** [comment]
Answering to myself after some research. Devs please correct me if I am again wrong, otherwise the topic can be closed.

I was using the wrong annotation, the EnsDb.Mmusculus.v79 is based on the GRCm38 (you get this info if you just run EnsDb.Mmusculus.v79 as a command in R after loading the library) and the reference for my dataset was GRCm39. For anyone running into a similar issue in the future, this is what I did to solve the problem:

I used AnnotationHub package to query the Annotation Hub for the available annotations for my genome and release:
```
library(AnnotationHub)
ah <- AnnotationHub() 
query(ah, c("Mus musculus", "release-104")) # I used release-104 because this was the release for the reference genome as reported in the cellranger web output, see first post
```
I chose the ID of the appropriate GTF file from the query output and built an Ensembl Annotation package using functions from the ensembldb library:

```
Gtf <- ah["AH92202"]
DbFile <- ensembldb::ensDbFromAH(Gtf)
edb <- EnsDb(DbFile)
```
Finally, I prepared the annotations object as in the Signac vignettes:

```
# get gene annotations
annotations <- GetGRangesFromEnsDb(ensdb = edb)
seqlevels(annotations) <- paste0('chr', seqlevels(annotations))
```

Cheers,
Marko

**Reply by leecon30** [reply]
You can also just use the tss.bed annotations made by cellranger atac/arc mkref. You can import them as a GrangesObject using readbed tool from genomation package. Then use it as tss coords for TSSenrichment. For custom genomes.
